add.go

```diff
@@ -0,0 +1,231 @@
+package ipfscluster
+
+import (
+	"context"
+	"errors"
+	"mime/multipart"
+	"net/url"
+	"strconv"
+
+	"github.com/ipfs/ipfs-cluster/api"
+	"github.com/ipfs/ipfs-cluster/importer"
+
+	"github.com/ipfs/go-ipfs-cmdkit/files"
+)
+
+func (c *Cluster) consumeLocalAdd(
```

Comment:
@ZenGround0 Why not have `consumeLocalAdd` and `consumeShardAdd` return `(string, error)` and not pass `args` to those two functions? It seems unnecessary to pass the state round when it could just be handled within `consumeImport`

Reply:
Actually, both `consumeLocalAdd` and `consumeShardAdd` could be slimmed down to just receive a `*api.NodeWithMeta` constructed outside of those two functions instead of inside.

Reply:
I agree you could do these things.  I went this route to make what is happening more explicit to the reader.  The string passed to `finish` has a totally different meaning depending on whether this is local or sharded.  Existing is probably overkill though and might over complicate things, I'll try to simplify.

Reply:
Yeah I get that this is the issue, @hsanjuan also did not like this.  I will refactor so that consume returns a string which I think appeals to both of your styles better. 

I'm not sure if I'm understanding your confusion properly but the reason we can't just send information into the consume function is that in general certain state (root cid/shard session id) must persist between calls to consume or finish.  True, we could read the outObj pointer after consume finished in consumeImport but then we need to read specific fields based on whether this is local or sharded call and that breaks the abstraction I was going for.